election-server/src/bin/indexer.rs

```diff
@@ -477,11 +524,23 @@ async fn node_process(
         let Some(block) = block else {
             return Err(anyhow!("Finalized block stream dropped"));
         };
+        let block_info = node
+            .get_block_info(block.block_hash)
```

Comment:
```suggestion
            .get_block_info(block.block_height)
```

queries by block height are much faster, especially for older blocks.